### ISSUE TITLE
Value of ElapsedTimeIdlingResource?

### ISSUE THREAD

**Issue by RobertLK** [chiuki/espresso-samples#9]
Been thinking a lot about this recently because I've seen it in a few projects and I'm struggling to see what it offers over a simple `Thread.sleep`. The only context I can find is [this blog post](http://blog.sqisland.com/2015/06/espresso-elapsed-time.html), which simply states "We could use SystemClock.sleep(60000), but that blocks the testing thread.". This is a strange justification since IdlingResources cause `onView` and `onIdle` to block the testing thread and this is wanted and expected behaviour for test code (tests are always run synchronously).
As far as I can tell using this IdlingResource is identical to using a sleep from a thread execution standpoint, but the code is more verbose and harder to understand. This seems to me like a misuse of IdlingResource and a testing anti-pattern, and it seems to be used in many codebases where a sleep would be simpler. Please let me know if there's some other execution property that makes this implementation advantageous.
Thanks.

**Comment by chiuki** [comment]
I agree with you. I wrote `ElapsedTimeIdlingResource` mostly as an exercise to understand how idling resources work, and there is no real use case. I wish I can retract it but it's been a few years so I'm not sure what's the best way to deal with it.

**Comment by RobertLK** [comment]
Thanks for replying
Definitely tricky with so many people using it, but it's not a big problem
if they've got it working and they're happy.
This is still a common challenge new developers face though, and this repo
comes high in search results, so adding more context here could be helpful.
Just a note in the code or readme to say that using sleep is usually fine
and may be the cleanest solution could save some confusion.
Up to you though, certainly not a major issue. 😎
